api/tests/DataPersister/ContentNodes/ColumnLayoutDataPersisterTest.php

```diff
@@ -0,0 +1,148 @@
+        $this->assertFalse($this->dataPersister->supports($this->contentNode, []));
+    }
+
+    public function testDoesNotSupportNonColumnLayout() {
+        $this->decoratedMock
+            ->method('supports')
+            ->willReturn(true)
+        ;
+
+        $this->assertFalse($this->dataPersister->supports([], []));
+    }
+
+    public function testDelegatesPersistToDecorated() {
+        // given
+        $this->decoratedMock->expects($this->once())
+            ->method('persist')
+        ;
+
+        // when
+        $this->dataPersister->persist($this->contentNode, []);
+
+        // then
+    }
+
+    public function testSetsRootFromParentOnCreate() {
+        // given
+        $this->decoratedMock->expects($this->once())->method('persist')->willReturnArgument(0);
+
+        // when
+        /** @var ColumnLayout $data */
+        $data = $this->dataPersister->persist($this->contentNode, ['collection_operation_name' => 'post']);
+
+        // then
+        $this->assertEquals($this->root, $data->root);
+    }
+
+    public function testCopyFromPrototypeOnCreate() {
+        // given
+        $this->decoratedMock->expects($this->once())->method('persist')->willReturnArgument(0);
+
+        // when
+        /** @var ColumnLayout $data */
+        $data = $this->dataPersister->persist($this->contentNode, ['collection_operation_name' => 'post']);
+
+        // then
+        $this->assertEquals($data->instanceName, $this->contentNode->prototype->instanceName);
+        $this->assertEquals($data->slot, $this->contentNode->prototype->slot);
+        $this->assertEquals($data->position, $this->contentNode->prototype->position);
+        $this->assertEquals($data->jsonConfig, $this->contentNode->prototype->jsonConfig);
+        $this->assertEquals($data->contentType, $this->contentNode->prototype->contentType);
+    }
+
+    public function testDoesNotOverrideDataOnCreate() {
+        // given
+        $this->contentNode->instanceName = 'testInstance';
+        $this->contentNode->slot = 'right';
+        $this->contentNode->position = 51;
+        $this->contentNode->jsonConfig = ['data' => 'value'];
+        $this->contentNode->contentType = new ContentType();
+
+        $this->decoratedMock->expects($this->once())->method('persist')->willReturnArgument(0);
+
+        // when
+        /** @var ColumnLayout $data */
+        $data = $this->dataPersister->persist($this->contentNode, ['collection_operation_name' => 'post']);
+
+        // then
+        $this->assertNotEquals($data->instanceName, $this->contentNode->prototype->instanceName);
+        $this->assertNotEquals($data->slot, $this->contentNode->prototype->slot);
+        $this->assertNotEquals($data->position, $this->contentNode->prototype->position);
+        $this->assertNotEquals($data->jsonConfig, $this->contentNode->prototype->jsonConfig);
+        $this->assertNotEquals($data->contentType, $this->contentNode->prototype->contentType);
```

Comment:
Ich hätte jetzt eher überprüft, dass die werte gleich sind wie auf $this->contentNode.
Kommt auf das gleicher heraus, macht aber den test strickter.